libs/vec/native/src/vec/headers/vec.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+#define EXPORT __attribute__((externally_visible,visibility("default")))
+
+#ifndef STRIDE_BYTES_LEN
```

Comment:
Nit: I would define this in the c file, as we don't need to "publish" it to other c code (and we have stride())

Reply:
Done. actually I moved these into separate functions, for now.